k2/csrc/array_ops.h

```diff
@@ -380,6 +406,40 @@ Array2<T> ToContiguous(const Array2<T> &src);
 template <typename T>
 bool Equal(const Array2<T> &a, const Array2<T> &b);
 
+/*
+  Index `src` with `indexes`, as in src[indexes].
+     @param [in] src   Array whose elements are to be read
+     @param [in] indexes  Indexes into `src`; must satisfy
+                       `0 <= indexes[i] < indexes.Dim()` if
+                       `allow_minus_one == true`,
+                       else -1 is also allowed and the corresponding
+                       output element will be zero.
+     @return  Returns an `Array1<T>` of dimension indexes.Dim(),
+               with `ans[i] = src[indexes[i]]` (or zero if
+               `allow_minus_one == true` and `indexes[i] == 0`).
+ */
+template <typename T>
+Array1<T> Index(const Array1<T> &src, const Array1<int32_t> &indexes,
+                bool allow_minus_one);
+
+
+/*
+  Index `src` with `indexes`, as in src[indexes].
+     @param [in] src   Array whose elements are to be read
+     @param [in] indexes  Indexes into `src`; must satisfy
+                       `0 <= indexes[i] < indexes.Dim0()` if
+                       `allow_minus_one == true`,
+                       else -1 is also allowed and the corresponding
+                       output element will be zero.
+     @return  Returns an `Array2<T>` of shape (indexes.Dim(), src.Dim1()),
+               with `ans[i,j] = src[indexes[i,j]]` (or zero if
+               `allow_minus_one == true` and `indexes[i] == 0`).
```

Comment:
and `indexes[i] == -1`? Add in the implementation seems we just check `indexes[i] < 0` instead of `indexes[i] == -1`